common/report.c

```diff
@@ -190,3 +190,42 @@ oe_result_t oe_get_target_info(
 done:
     return result;
 }
+
+oe_result_t oe_get_target_info_v2(
+    const uint8_t* report,
+    size_t report_size,
+    void** target_info_buffer,
+    size_t* target_info_size)
+{
+    oe_result_t result = OE_FAILURE;
+
+    *target_info_buffer = NULL;
```

Comment:
Is returning an allocated buffer and a size as two out parameters the recommended API style?
Why have an opaque type where the first field is the size of the structure?
struct oe_target_info_t {
    size_t size;
    uint8_t data[1];
};

Reply:
@anakrish This is an agreed upon v2 API. If we want to change it again then that would become a v3 API

Reply:
@gupta-ak fixed

Reply:
crashing when the caller passes in an illegal argument is good behavior.  Null checking the params leads to bugs, since callers don't necessarily check return values.   Hence the API guidance in Microsoft API reviews is that crashing is better behavior.